src/libraries/System.Net.Http.Json/tests/FunctionalTests/HttpContentJsonExtensionsTests.cs

```diff
@@ -201,6 +182,7 @@ public async Task EnsureDefaultJsonSerializerOptionsAsync()
         [InlineData("application/foo+json")] // Structured Syntax Json Suffix
         [InlineData("application/foo+Json")]
         [InlineData("appLiCaTiOn/a+JsOn")]
+        [InlineData("application/json-custom")]
         public async Task TestValidMediaTypes(string mediaType)
```

Comment:
Since all media types are acceptable now, we should rename this to "TestVariousMediaTypes" or something like that. Also add a completely meaningless media type like "asdf/ghjk" or whatever.